showkase-processor/src/main/java/com/airbnb/android/showkase/processor/ShowkaseProcessor.kt

```diff
@@ -127,6 +135,92 @@ class ShowkaseProcessor @JvmOverloads constructor(
             }.flatten().mapNotNull { it }.toSet()
     }
 
+    private fun processCustomAnnotation(
+        roundEnvironment: XRoundEnv,
+        annotation: XTypeElement? = null
+    ): Set<ShowkaseMetadata.Component> {
+        val supportedTypes = mutableListOf<String>()
+        if (annotation != null) supportedTypes.add(annotation.qualifiedName)
+        val components = mutableSetOf<ShowkaseMetadata.Component>()
+
+        supportedTypes.map { supportedType ->
+            val annotatedElements = roundEnvironment.getElementsAnnotatedWith(supportedType)
+            annotatedElements.map { annotatedElement ->
+                if (!showkaseValidator.checkElementIsAnnotationClass(annotatedElement)) {
+
+                    showkaseValidator.validateComponentElementOrSkip(
+                        element = annotatedElement,
+                        annotationName = supportedType,
+                    )
+                    components.addAll(
+                        getShowkaseMetadataFromCustomAnnotation(
+                            element = annotatedElement,
+                            showkaseValidator = showkaseValidator,
+                            supportedType.getCustomAnnotationSimpleName(),
+                        ).toSet()
+                    )
+                }
+            }
+        }
+        return components
+    }
+
+    private fun String.getCustomAnnotationSimpleName(): String {
+        return this.split(".").last()
+    }
+
+    private fun processCustomAnnotationFromClasspath(roundEnvironment: XRoundEnv): Set<ShowkaseMetadata.Component> {
+        // In this function we are checking generated classpath for MultiPreview codegen annotations.
+        // We also check the current module if there is any composables that are annotated with the qualified name
+        // from the annotation from classpath. We use the fields from the classpath annotation to build
+        // common data for the ShowkaseMetadata.
+
+        // Supported annotations from classpath
+        val supportedCustomPreview = environment.getTypeElementsFromPackage(CODEGEN_PACKAGE_NAME)
+            .flatMap { it.getEnclosedElements() }.mapNotNull {
+                when (
+                    val annotation = it.getAnnotation(ShowkaseMultiPreviewCodegenMetadata::class)
+                ) {
+                    null -> null
+                    else -> annotation.value
+                }
+            }
+        val components = mutableSetOf<ShowkaseMetadata.Component>()
+
+        supportedCustomPreview
+            .mapIndexed { index: Int, customPreviewMetadata: ShowkaseMultiPreviewCodegenMetadata ->
+                roundEnvironment
+                    .getElementsAnnotatedWith(customPreviewMetadata.supportTypeQualifiedName)
+                    .mapIndexed elementRoot@{ elementIndex, xElement ->
+
+                        if (showkaseValidator.checkElementIsAnnotationClass(xElement)) {
```

Comment:
The `xElement` here is a custom annotation like `@CombinedPreview`. Under what circumstances would that annotation be annotated to an annotation class itself?

Reply:
I think my thought with this was that if you had a case of something like 

```
@Preview
@CombinedPreview
fun SomePreview()
```
or 
```
@FontPreview
@CombinedPreview
fun SomePreview()
```

Reply:
hmm, when thinking about it that is a good point! Maybe I can remove this check actually 🤔

Reply:
Locally at least it worked without this check. I added another module as well to test it. Pushed the changes now :)